includes/API/Packs/ApiLabkiPacksPreview.php

```diff
@@ -0,0 +1,496 @@
+ *     "total_pages": 25,
+ *     "conflicts": 3,
+ *     "dependency_issues": 0
+ *   },
+ *   "packs": [
+ *     {
+ *       "name": "Advanced Imaging",
+ *       "action": "install",
+ *       "version": "1.0.0",
+ *       "auto_included": false,
+ *       "dependencies": [
+ *         {
+ *           "name": "Hardware Setup",
+ *           "status": "included",
+ *           "action": "update"
+ *         },
+ *         {
+ *           "name": "Calibration Tools",
+ *           "status": "auto_selected",
+ *           "action": "install"
+ *         }
+ *       ],
+ *       "pages": [
+ *         {
+ *           "name": "Processing",
+ *           "default_title": "Imaging/Processing",
+ *           "conflict": {
+ *             "type": "title_exists",
+ *             "existing_page_id": 456,
+ *             "suggested_title": "Imaging/Advanced_Processing"
+ *           }
+ *         }
+ *       ]
+ *     }
+ *   ]
+ * }
+ * ```
+ *
+ * @ingroup API
+ */
+final class ApiLabkiPacksPreview extends PackApiBase {
+
+	/** @inheritDoc */
+	public function __construct( \ApiMain $main, string $name ) {
+		parent::__construct( $main, $name );
+	}
+
+	/** Execute the API request. */
+	public function execute(): void {
+		$params = $this->extractRequestParams();
+
+		// Get parameters
+		$repoId = $params['repo_id'] ?? null;
+		$repoUrl = $params['repo_url'] ?? null;
+		$refId = $params['ref_id'] ?? null;
+		$ref = $params['ref'] ?? null;
+		$operationsJson = $params['operations'] ?? '[]';
+
+		wfDebugLog( 'labkipack', "ApiLabkiPacksPreview::execute() repo_id={$repoId}, ref={$ref}" );
+
+		// Validate: only one repo identifier
+		if ( $repoId !== null && $repoUrl !== null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-repo-multiple-identifiers',
+				'multiple_identifiers'
+			);
+		}
+
+		// Validate: only one ref identifier
+		if ( $refId !== null && $ref !== null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-ref-multiple-identifiers',
+				'multiple_identifiers'
+			);
+		}
+
+		// Validate: repo is required
+		if ( $repoId === null && $repoUrl === null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-repo-required',
+				'missing_repo'
+			);
+		}
+
+		// Validate: ref is required
+		if ( $refId === null && $ref === null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-ref-required',
+				'missing_ref'
+			);
+		}
+
+		// Parse operations JSON
+		$operations = json_decode( $operationsJson, true );
+		if ( !is_array( $operations ) || empty( $operations ) ) {
+			$this->dieWithError(
+				'labkipackmanager-error-invalid-operations',
+				'invalid_operations'
+			);
+		}
+
+		// Validate each operation
+		foreach ( $operations as $op ) {
+			if ( !isset( $op['action'] ) || !isset( $op['pack_name'] ) ) {
+				$this->dieWithError(
+					'labkipackmanager-error-invalid-operation-format',
+					'invalid_operation_format'
+				);
+			}
+			if ( !in_array( $op['action'], [ 'install', 'update', 'remove' ], true ) ) {
```

Comment:
possible actions should not be defined inline, they should be an enum or a const defined publicly and used universally.